ext/mysql2/extconf.rb

```diff
@@ -105,17 +105,19 @@ def asplode(lib)
   '-Wno-missing-field-initializers', # gperf generates bad code
   '-Wno-missing-variable-declarations', # missing symbols due to ruby native ext initialization
   '-Wno-padded', # mysql :(
+  '-Wno-reserved-id-macro', # rubby :(
   '-Wno-sign-conversion', # gperf generates bad code
   '-Wno-static-in-inline', # gperf generates bad code
   '-Wno-switch-enum', # result.c -- enum_field_types (when not fully covered, e.g. mysql 5.6+)
   '-Wno-undef', # rubinius :(
+  '-Wno-unreachable-code', # rubby :(
   '-Wno-used-but-marked-unused', # rubby :(
 ]
 
 if ENV['CI']
   wishlist += [
     '-Werror',
-    '-fsanitize=address',
+    '-fsanitize=address -lasan',
```

Comment:
This could be needed because it's a "plugin" to the compiler chain on some (all?) systems. Though, I've definitely used `-fsanitize=address` on OSX and not had to link to `libasan` myself so I'm not really sure...